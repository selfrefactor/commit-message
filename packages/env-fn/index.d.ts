type Modes = 'special' | 'local'

export function envFn(mode?: Modes, cwd?: string): string[]