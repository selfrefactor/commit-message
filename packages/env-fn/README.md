# Env-fn

Load .env configuration

## Usage

```
require("env-fn")()

require("env-fn")(".env")
require("env-fn")("special")
```

## Typing

`require("env-rn")(fileName: string)`

Searches for `fileName` starting from `process.cwd` directory and moving up.

If it finds it, it loads environment variables in the process.

Only the first request load variables; the following calls do nothing.

## Default value

- fileName - `.env`

## Pin to different directory

Default directory is `process.cwd()`
To apply local directory as start point use 'dirname' as argument, which is upgrade upon `special` behaviour.

## Forbid use through env flag

If your `process.env.ENV_FLAG` is different than `undefined` or `'true'`, then no environment variables will be loaded with this library.
