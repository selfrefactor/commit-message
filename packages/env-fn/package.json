{
	"dependencies": {
		"rambda": "6.4.0"
	},
	"name": "env-fn",
	"version": "2.2.0",
	"typings": "index.d.ts",
	"description": "Load environment variables from from local `.env` file",
	"main": "./src/env-fn.js",
	"repository": "https://github.com/selfrefactor/services/tree/master/packages/env-fn",
	"author": "Dejan Toteff <deyan8284@gmail.com>",
	"license": "MIT",
	"devDependencies": {}
}
