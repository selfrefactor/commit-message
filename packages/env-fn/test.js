const { envFn } = require('./envFn')

// SPECIAL
// ============================================
const envs = envFn('special')

//  LOCAL WITH CWD
// ============================================
// const envs = envFn('local', `${__dirname}/test_folder`)

// RESULTS
// ============================================
console.log('=============')
console.log({ envs })
console.log('=============')
console.log({
  redis : process.env.REDIS_URL,
  foo   : process.env.foo,
})
