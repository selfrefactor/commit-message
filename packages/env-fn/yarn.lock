# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


rambda@6.4.0:
  version "6.4.0"
  resolved "https://registry.yarnpkg.com/rambda/-/rambda-6.4.0.tgz#bf080047b03d8d2c72e3ce5c17c603517a65749a"
  integrity sha512-aBoE7YrGQP1xnxjyg5fhupfS+FTrDVjZMtwpezl15RKLNnKLOYDfjx/0xDE+8GyWbbZfODRkAOZmnpvxrAONbQ==
