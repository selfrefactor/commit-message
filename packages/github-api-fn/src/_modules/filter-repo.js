const axios = require('axios')
const dayjs  = require('dayjs')
const { path, defaultTo }  = require('rambdax')

const DAYS_LIMIT = 370
const FILE = 'package.json'
const FALLBACK = {
  pass       : false,
  updateDate : undefined,
  updateDiff : undefined,
}

function dateDiff(updateDate){
  const now = dayjs(Date.now())
  const past = dayjs(updateDate)

  return now.diff(past, 'day')
}

async function filterRepo(repo, daysLimitInput){
  const daysLimit = defaultTo(DAYS_LIMIT, daysLimitInput)

  const token = process.env.GITHUB
  if (!token) throw new Error('!token')
  if (!repo.includes('/')) throw new Error(`wrong repo input - ${ repo }`)
  const url = `https://api.github.com/repos/${ repo }/commits?path=${ FILE }&page=1&per_page=1`
  const { data } = await axios({
    method  : 'get',
    url,
    timeout : 20000,
    headers : { Authorization : `token ${ process.env.GITHUB }` },
  })
  if (data.length === 0){
    return FALLBACK
  }
  const updateDate = path('commit.committer.date', data[ 0 ])

  const diff = dateDiff(updateDate)

  return {
    pass       : diff < daysLimit,
    updateDate,
    updateDiff : diff,
  }
}

exports.filterRepo = filterRepo
exports.dateDiff = dateDiff