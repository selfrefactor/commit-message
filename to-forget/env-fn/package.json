{
	"dependencies": {
		"rambda": "6.2.0"
	},
	"name": "env-fn",
	"version": "2.1.1",
	"typings": "index.d.ts",
	"description": "Load environment variables from from local `.env` file",
	"main": "envFn.js",
	"repository": "https://github.com/selfrefactor/services/tree/master/packages/env-fn",
	"author": "Dejan Toteff <deyan8284@gmail.com>",
	"license": "MIT",
	"devDependencies": {}
}
