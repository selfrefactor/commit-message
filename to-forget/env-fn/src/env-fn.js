const { head, split, replace } = require('rambda')
const { readFileSync, existsSync } = require('fs')
const { resolve } = require('path')
const {getEnvPath } = require('./get-env-path')

const applyEnvFn = filePath => {
  const allEnv = readFileSync(filePath, 'utf8')
  const toReturn = []

  allEnv
    .split('\n')
    .filter(Boolean)
    .map(line => {
      const key = head(split('=', line)).trim()
      const val = replace(
        `${ key }=`, '', line
      ).trim()

      process.env[ key ] = val
      toReturn.push(`${ key }=${ val }`)
    })
  console.log(`\u2713    Environment variables loaded from '${ filePath }'`)

  return toReturn.join(';')
}

const envFn = (mode = 'local', cwd = process.cwd()) => {
  if (process.env.ENV_FLAG === undefined){
    process.env.ENV_FLAG = 'true'
  } else {
    return
  }

  const initialFilePath = resolve(cwd, '.env')

  if (existsSync(initialFilePath) && mode === 'local'){
    return applyEnvFn(initialFilePath)
  }

  const filePath = getEnvPath(mode === 'special', cwd)

  if (filePath === undefined || existsSync(filePath) === false){
    throw new Error('Such env filepath does not exist !!!')
  }

  return applyEnvFn(filePath)
}

exports.envFn = envFn
