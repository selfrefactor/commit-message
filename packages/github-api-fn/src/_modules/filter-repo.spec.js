import { filterRepo } from './filter-repo'
import {envFn} from 'env-fn'
envFn('special')

test('happy', async () => {
  expect(
    await filterRepo('selfrefactor/rambda')
  ).toMatchSnapshot()
})

test('no package.json', async () => {
  expect(
    await filterRepo('selfrefactor/commit')
  ).toMatchSnapshot()
})