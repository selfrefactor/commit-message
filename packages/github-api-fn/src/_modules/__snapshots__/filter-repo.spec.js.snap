// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Object {
  "pass": true,
  "updateDate": "2020-12-21T20:57:54Z",
  "updateDiff": 5,
}
`;

exports[`no package.json 1`] = `
Object {
  "pass": false,
  "updateDate": undefined,
  "updateDiff": undefined,
}
`;
